src/app/static/src/js/session.js

```diff
@@ -0,0 +1,40 @@
+export const session = function(){
+   const RUNNING_REPORT_STATUS_PREFIX = "runningReportStatus_";
+
+   const RUNNING_STATUS = "runningStatus";
+   const RUNNING_KEY = "runningKey";
+   const NEW_VERSION_FROM_RUN = "newVersionFromRun";
+   const NEW_VERSION_DISPLAY_NAME = "newVersionDisplayName";
+
+   const RUNNING_REPORT_STATUS_KEYS = [RUNNING_STATUS, RUNNING_KEY, NEW_VERSION_FROM_RUN, NEW_VERSION_DISPLAY_NAME];
+
+   function getItem(key) {
+       return window.sessionStorage.getItem(key);
+   }
+
+   function setItem(key, value) {
+       window.sessionStorage.setItem(key, value);
```

Comment:
If `value` is null here it gets coerced to a string and saved as "null", so when you read back out a null value you actually get a non-null string (meaning the page thinks values exist when they don't - e.g. try running a report that errors, then refreshing the page - page thinks there is a new version even though there isn't.) I think check for null here and remove the key if the value is null. Then when reading back `getItem(key)` will return null as expected.

Reply:
Unfortunately can't think of a good way to test this other than by selenium test!